src/userbase-js/auth.js

```diff
@@ -246,37 +265,36 @@ const displayShowKeyModal = (seedString, rememberMe) => new Promise(resolve => {
   closeButton.onclick = hideShowKeyModal
 })
 
-const signUp = async (username, password, email, profile, showKeyHandler, rememberMe = false) => {
+const signUp = async (username, password, email, profile, showKeyHandler, rememberMe = false, passwordBasedKeyRecoveryEnabled = true) => {
```

Comment:
I think `backupKey` might be a better param name than `passwordBasedKeyRecoveryEnabled`. The password-based bit is an implementation detail.

Also, a minor thing, but should `showKeyHandler` be the last param since it's optional?

Reply:
I was thinking we'd want to expose that implementation detail here because it significantly alters the security properties of the service. Without any other knowledge aside from looking at the API, no developer will be inclined to set `backupKey` to false (or look into setting it to false) because keeping it true feels like the safer thing to do, but `passwordBasedKeyRecoveryEnabled` makes it a bit clearer how the developer can think through the choice.

I'm in to assume developers know what they're doing and go with `backupKey`, it would certainly make it cleaner

> Also, a minor thing, but should showKeyHandler be the last param since it's optional?

All parameters after password are optional

Reply:
I don't think `passwordBasedKeyRecoveryEnabled` helps though. And since it's enabled by default, we're choosing to make this our default security posture. Developers won't be opting-into "less security" (because this is the default baseline that we'd claim), but they have the option to opt-in "more security" (at the cost of inconveniencing the users) if they really want to. I think the best way to educate here is through the docs. We can have a section about security & privacy, and explain our claims, all the options available to the developers, and the things they should avoid (such as third-party analytics on their app, etc).

Reply:
This is fair. Agreed.

Reply:
Slight modification proposal: `backUpKey` rather than `backupKey`. The former is an action/verb phrase which fits that this is a boolean. The latter is a thing/noun which fits an object

Reply:
Yep, sounds good. Let's go with backUpKey.